code/game/objects/structures/mirror.dm

```diff
@@ -36,44 +73,149 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
 
 MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror/broken, 28)
 
-/obj/structure/mirror/attack_hand(mob/user, list/modifiers)
+/obj/structure/mirror/attack_hand(mob/living/carbon/human/user)
 	. = ..()
-	if(.)
-		return TRUE
-	if(broken || !Adjacent(user))
-		return TRUE
 
-	if(!ishuman(user))
+	if(. || !ishuman(user) || broken)
 		return TRUE
-	var/mob/living/carbon/human/hairdresser = user
 
-	//handle facial hair (if necessary)
-	if(hairdresser.gender != FEMALE)
-		var/new_style = tgui_input_list(user, "Select a facial hairstyle", "Grooming", GLOB.facial_hairstyles_list)
+	if(!user.can_perform_action(src, FORBID_TELEKINESIS_REACH) && !magical_mirror)
+		return TRUE //no tele-grooming (if nonmagical)
+
+	return what_is_your_wish_masta(user)
+
+/obj/structure/mirror/proc/what_is_your_wish_masta(mob/living/carbon/human/user)
+	var/pick = show_radial_menu(user, src, mirror_options, user, radius = 36, require_near = TRUE)
+	if(!pick)
+		return TRUE //get out
+
+	switch(pick)
+		if(CHANGE_HAIR)
+			change_hair(user)
+		if(CHANGE_BEARD)
+			change_beard(user)
+		if(CHANGE_RACE)
+			change_race(user)
+		if(CHANGE_SEX) // sex: yes
+			change_sex(user)
+		if(CHANGE_NAME)
+			change_name(user)
+		if(CHANGE_EYES)
+			change_eyes(user)
+
+	return what_is_your_wish_masta(user)
+
+/obj/structure/mirror/proc/change_beard(mob/living/carbon/human/beardresser)
```

Comment:
can this be `beard_dresser` instead the `d` pulling double-duty irks me slightly